Anlab.Mvc/Controllers/TestItemsController.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using Anlab.Core.Domain;
+using Anlab.Core.Data;
+using Microsoft.AspNetCore.Authorization;
+using AnlabMvc.Models.Roles;
+
+
+namespace AnlabMvc.Controllers
+{
+    [Authorize(Roles = RoleCodes.Admin)]
+    public class TestItemsController : ApplicationController
+    {
+        private readonly ApplicationDbContext _context;
+
+        public TestItemsController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        // GET: TestItems
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.TestItems.ToListAsync());
+        }
+
+        // GET: TestItems/Details/5
+        public async Task<IActionResult> Details(string id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var testItem = await _context.TestItems
+                .SingleOrDefaultAsync(m => m.Id == id);
+            if (testItem == null)
+            {
+                return NotFound();
+            }
+
+            return View(testItem);
+        }
+
+        // GET: TestItems/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: TestItems/Create
+        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
+        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Analysis,Category,Group,Public,Notes")] TestItem testItem)
+        {
+            testItem.Id = testItem.Id.ToUpper();
+            if (_context.TestItems.Any(t => t.Id == testItem.Id))
+            {
+                ErrorMessage = "Id already in use";
+                return View();
+            }
+            if (ModelState.IsValid)
+            {
+                
```

Comment:
@laholstege When @jSylvestre says add a message, he means there is a line in our layout page that shows any value added into @TempData["Message"].  So on save you would to TempData["Message"] = "Test XYZ was saved successfully", and then when the user was redirected to index it'd show that they saved properly.  Same thing with other actions